app/src/main/java/de/rampro/activitydiary/search/ActivityDiarySuggestionProvider.java

```diff
@@ -126,10 +146,37 @@ public Cursor query(@NonNull Uri uri, @Nullable String[] projection, @Nullable S
             default:
                 /* empty */
         }
+        /*
+            If no query is typed, then show recent suggestions
+         */
+        if (query == null) {
+            LocalDBHelper mOpenHelper = new LocalDBHelper(getContext());
+            SQLiteDatabase db = mOpenHelper.getReadableDatabase();
+            String sql;
+            sql = "SELECT " + ActivityDiaryContract.DiarySuggestion.SUGGESTION + " FROM " +
+                    ActivityDiaryContract.DiarySuggestion.TABLE_NAME +
+                    " ORDER BY " + ActivityDiaryContract.DiarySuggestion.lAST_CHANGED + " DESC";
```

Comment:
yes, but I am confident, that as soon as a user has searched 2^64 searches, 1. you an me are not involved in the development anymore due to biological reasons and 2. it will at least run on a 128 bit machine where the native integer type of SQLite was changed to 16 bytes :grin:

-> I was tending to elaborate on changing only those things that are important for you, but I tried to keep the comment above short. So now I can still answer with this useless comment here :smiley: (or is it important for us to also have some fun here? clearly yes.) - I cannot emphasize enough, that you should not take me too serious. And if it's too much and you don't have fun or it feel as I would make fun of you, please tell me, that is not my intention. I just try to not be too boring, while still letting you know my opinion in the comments. And I don't have the expectation that we finally agree on every small detail. What counts is functionality and from that point of view your PR is ready to be merged from the time you submitted it, because it IS an improvement.